### DISCUSSION TITLE
Time for a 1.13 release

### DISCUSSION THREAD

**Discussion by gjasny** [resiprocate/resiprocate#232] [General]
Hello @sgodin, @dpocock,

quite a few fixes and improved modern C++11 compatibility sit in the master branch. How about a new 1.13 release?

Thanks,
Gregor

**Comment by tomy2105** [comment]
Yes, definitely would be nice to know when the 1.13 is about to be out.... :)

**Comment by sgodin** [comment]
I would also support a new release soon.  We need Daniel on board too,
since he manages the linux release side of things.  I know he's been
working on recon, so he may want to delay until that is done, not 100% sure.

@dpocock - Daniel what are your thoughts?

Scott

**Comment by alienpenguin** [comment]
Is there any update on this?

**Comment by mbeaulie** [comment]
Yes, an update would be appreciated. We are looking for OpenSSL 3 compatibility and I think that is not part of 1.12,

**Comment by searlers** [comment]
Some kind of definite timeline/plan would be helpful. 1.13 release imminent was announced "2020-11-10 - reSIProcate 1.13 coming soon" quite a while ago in the scheme of things. It might help people to assist in the right areas if they knew the specifics of blockers for release

**Comment by vak-oskh** [comment]
Hello @sgodin,  @dpocock,

An update for "2020-11-10 - reSIProcate 1.13 coming soon" would be appreciated. Do you have any roadmap for that?
Thanks in advance.

**Comment by sgodin** [comment]
Sorry guys @dpocock has been in charge of resip releases for sometime now, as he is aware of all the in's and out's of releasing a linux distribution package.  Unfortunately he has been missing from this project for a while now, and has been unresponsive.  

I would be happy to have someone else take over this from him.  Please contact me directly if you have interest in this.

Scott

**Reply by Lastique** [reply]
Is there noone able to even just push a git tag? It appears like @dpocock is no longer active, so the release procedure needs to be updated. That is unless this project is effectively dead (which would be a shame).

**Comment by sgodin** [comment]
The project is not dead, but we definitely need more volunteers to help
with the project.

Also, there is more to do than just pushing a tag.  Release notes need to
be generated, we need to make sure the new CMake build system is working,
and that we have good instructions for linux and windows users.  Note:
CMake currently doesn't work for windows users, but the checked VS
solutions and projects do work, so that is good.  To accomplish this we
need to get the CI working, which is also currently broken.  Please let me
know if you are willing to help out.

**Reply by mykhailopopivniak** [reply]
Is there an updated checklist or roadmap for tasks needed to complete the new release? A clear outline would make it easier for contributors to know where they can help.

**Comment by sgodin** [comment]
@mykhailopopivniak Thank you for asking and taking an interest.  :)

Daniel Pocock (@dpocock) put this .txt file together back in 2020 - hopefully it is complete:
https://github.com/resiprocate/resiprocate/blob/master/RELEASE-PROCESS.txt

**Comment by perseoGI** [comment]
Hi, as a maintainer of Conan Center Index, we have a blocked PR https://github.com/conan-io/conan-center-index/pull/20465 that requires a new tag.

Is there any progress?
Thank you in advance 🐸

**Comment by sgodin** [comment]
I have started to dedicate some time to getting resip's new CMake build
system ready for a release.  When this is complete, I hope to get a 1.13
release available.

**Comment by sgodin** [comment]
I should be creating a branch for a 1.13 release very soon.  I'm planning on releasing this via GitHub releases.  

Daniel Pocock use to manage pushing releases to be distributed as an official Debian package.  I do NOT plan to do this, as I have very little experience in this area.  Daniel has left some notes on what to do for this, so if anyone in the community is interested in helping out here, then that would be appreciated.  Otherwise, we will at least have an official source code release. :)

I have examined all the outstanding PRs and merged / closed appropriately.  If there are any other changes desired to make this release, let's discuss that here.

Best Regards,
Scott Godin
http://www.sipspectrum.com

**Comment by sgodin** [comment]
One other note:  I would love to get a windows build setup in the GitHub actions.  If anyone has cycles to tackle this or help out, please let me know.  Any help is greatly appreciated.

**Comment by sgodin** [comment]
Hello fellow reSIPer's.  I have reached a point where I think we might be ready for 1.13 release.  I have created a branch for it:

https://github.com/resiprocate/resiprocate/tree/resiprocate-1.13

and added a ChangeLog file:

https://github.com/resiprocate/resiprocate/blob/resiprocate-1.13/CHANGELOG.md

Please kick the tires with this branch and if there is no negative feed back, I will get it listed as our official 1.13 release! 

Happy Holidays!

Scott Godin
http://www.sipspectrum.com/

**Comment by sgodin** [comment]
resiprocate 1.13.0 Beta 1 is now available for download here:
https://github.com/resiprocate/resiprocate/releases/tag/resiprocate-1.13.0_beta1

**Reply by perseoGI** [reply]
Great!
When are you expecting to release the final one?

**Comment by sgodin** [comment]
I have no specific date in mind.  I'd like to give people the opportunity to play with the beta version to see what other changes might be needed for a final release.  That being said, I would hope to have a final 1.13.0 release sometime in early to mid January.

I see that @gjasny has started to look at things, so that's great, and I would encourage others to do the same.
